esi_leap/objects/lease.py

```diff
@@ -132,14 +202,29 @@ def cancel(self):
                         parent_lease = Lease.get(self.parent_lease_uuid)
                         resource.set_lease(parent_lease)
 
-                self.status = statuses.DELETED
-                self.expire_time = datetime.datetime.now()
+                resource._node_name = resource.get_name()
+                resource._provision_state \
+                    = resource._get_node_attr('provision_state')
+                resource._power_state = resource._get_node_attr('power_state')
```

Comment:
I don't think we should be calling a `_` method; maybe add `get_*_state` methods?